File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ShipTo.cs
#region "Imports"

using System;
using PayPal.Payments.Common.Utility;
using PayPal.Payments.Common.Exceptions;

#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for shipping address information
	/// </summary>
	/// <remarks>Shipping address is destination address information.</remarks>
	/// <example>
	/// <para>Following example shows how to use ShipTo.</para>
	/// <code lang="C#" escaped="false">
	///  .................
	///  //Inv is the Invoice object.
	///  .................
	///  
	///	//Set the Shipping Address details.
	///	ShipTo Ship = new ShipTo();
	///	Ship.ShipToStreet = "685A E. Middlefield Rd.";
	///	Ship.ShipToStree2 = "Apt. #2";
	///	Ship.ShipToZip = "94043";
	///	Inv.ShipTo = Ship;
	///	.................
	/// </code>
	/// <code lang="Visual Basic" escaped="false">
	///  .................
	///  'Inv is the Invoice object.
	///  .................
	///  
	///	'Set the Shipping Address details.
	///	Dim Ship As ShipTo = New ShipTo
	///	Ship.ShipToStreet = "685A E. Middlefield Rd."
	///	Ship.ShipToStree2 = "Apt. #2";
	///	Ship.ShipToZip = "94043"
	///	Inv.ShipTo = Ship
	///	.................
	/// </code>
	/// </example>
	public sealed class ShipTo : Address
	{
		#region "Member Variables"

		/// <summary>
		/// Shipping method.
		/// </summary>
		private String mShipMethod;

		/// <summary>
		/// Shipping carrier
		/// </summary>
		private String mShipCarrier;

		/// <summary>
		/// Ship from zip
		/// </summary>
		private String mShipFromZip;

		#endregion

		#region "Properties"

		/// <summary>
		/// Gets, Sets shipping city 
		/// </summary>
		/// <remarks>
		/// Shipping city
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOCITY</code>
		/// </remarks>
		public String ShipToCity
		{
			get { return base.AddressCity; }
			set { base.AddressCity = value; }
		}

		/// <summary>
		/// Gets, Sets shipping country
		/// </summary>
		/// <remarks>
		/// Shipping country
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOCOUNTRY</code>
		/// </remarks>
		public String ShipToCountry
		{
			get { return base.AddressCountry; }
			set { base.AddressCountry = value; }
		}

		/// <summary>
		/// Gets, Sets shipping email
		/// </summary>
		/// <remarks>
		/// Shipping email
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOEMAIL</code>
		/// </remarks>
		public String ShipToEmail
		{
			get { return base.AddressEmail; }
			set { base.AddressEmail = value; }
		}

		/// <summary>
		/// Gets, Sets shipping first name
		/// </summary>
		/// <remarks>
		/// Shipping first name
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOFIRSTNAME</code>
		/// </remarks>
		public String ShipToFirstName
		{
			get { return base.AddressFirstName; }
			set { base.AddressFirstName = value; }
		}

		/// <summary>
		/// Gets, Sets shipping last name
		/// </summary>
		/// <remarks>
		/// Shipping last name
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOLASTNAME</code>
		/// </remarks>
		public String ShipToLastName
		{
			get { return base.AddressLastName; }
			set { base.AddressLastName = value; }
		}

		/// <summary>
		/// Gets, Sets shipping middle name
		/// </summary>
		/// <remarks>
		/// Shipping middle name
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOMIDDLENAME</code>
		/// </remarks>
		public String ShipToMiddleName
		{
			get { return base.AddressMiddleName; }
			set { base.AddressMiddleName = value; }
		}

		/// <summary>
		/// Gets, Sets shipping phone2
		/// </summary>
		/// <remarks>
		/// Shipping phone 2
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOPHONE2</code>
		/// </remarks>
		public String ShipToPhone2
		{
			get { return base.AddressPhone2; }
			set { base.AddressPhone2 = value; }
		}

		/// <summary>
		/// Gets, Sets shipping phone
		/// </summary>
		/// <remarks>
		/// Shipping phone
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOPHONE</code>
		/// </remarks>
		public String ShipToPhone
		{
			get { return base.AddressPhone; }
			set { base.AddressPhone = value; }
		}

		/// <summary>
		/// Gets, Sets shipping state
		/// </summary>
		/// <remarks>
		/// Shipping state
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOSTATE</code>
		/// </remarks>
		public String ShipToState
		{
			get { return base.AddressState; }
			set { base.AddressState = value; }
		}

		/// <summary>
		/// Gets, Sets shipping street
		/// </summary>
		/// <remarks>
		/// Shipping street
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOSTREET</code>
		/// </remarks>
		public String ShipToStreet
		{
			get { return base.AddressStreet; }
			set { base.AddressStreet = value; }
		}

		/// <summary>
		/// Gets, Sets shipping street2
		/// </summary>
		/// <remarks>
		/// Shipping street 2.
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOSTREET2</code>
		/// </remarks>
		public String ShipToStreet2
		{
			get { return base.AddressStreet2; }
			set { base.AddressStreet2 = value; }
		}

		/// <summary>
		/// Gets, Sets shipping zip
		/// </summary>
		/// <remarks>
		/// Ship to postal code (called ZIP code in the USA).
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPTOZIP</code>
		/// </remarks>
		public String ShipToZip
		{
			get { return base.AddressZip; }
			set { base.AddressZip = value; }
		}

		/// <summary>
		/// Gets, Sets shipping method
		/// </summary>
		/// <remarks>
		/// Shipping method
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPMETHOD</code>
		/// </remarks>
		public String ShipMethod
		{
			get { return mShipMethod; }
			set { mShipMethod = value; }
		}

		/// <summary>
		/// Gets, Sets shipping carrier
		/// </summary>
		/// <remarks>
		/// Shipping carrier
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPCARRIER</code>
		/// </remarks>
		public String ShipCarrier
		{
			get { return mShipCarrier; }
			set { mShipCarrier = value; }
		}

		/// <summary>
		/// Gets, Sets ship from zip
		/// </summary>
		/// <remarks>
		/// Ship from postal code (called ZIP code in the USA).
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>SHIPFROMZIP</code>
		/// </remarks>
		public String ShipFromZip
		{
			get { return mShipFromZip; }
			set { mShipFromZip = value; }
		}

		#endregion

		#region "Constructors"

		/// <summary>
		/// Constructor
		/// </summary>
		/// <remarks>Shipping address is destination address information.</remarks>
		/// <example>
		/// <para>Following example shows how to use ShipTo.</para>
		/// <code lang="C#" escaped="false">
		///  .................
		///  //Inv is the Invoice object.
		///  .................
		///  
		///	//Set the Shipping Address details.
		///	ShipTo Ship = new ShipTo();
		///	Ship.ShipToStreet = "685A E. Middlefield Rd.";
		///	Ship.ShipToStree2 = "Apt. #2";
		///	Ship.ShipToZip = "94043";
		///	Inv.ShipTo = Ship;
		///	.................
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///  .................
		///  'Inv is the Invoice object.
		///  .................
		///  
		///	'Set the Shipping Address details.
		///	Dim Ship As ShipTo = New ShipTo
		///	Ship.ShipToStreet = "685A E. Middlefield Rd."
		///	Ship.ShipToStree2 = "Apt. #2";
		///	Ship.ShipToZip = "94043"
		///	Inv.ShipTo = Ship
		///	.................
		/// </code>
		/// </example>
		public ShipTo()
		{
		}

		#endregion

		#region "Utility function"

		/// <summary>
		/// This method copies the common contents
		/// from shipping to billing address.
		/// </summary>
		/// <returns>Billing Address object</returns>
		/// <remarks>This method can be used to 
		/// populate the shipping addresses directly 
		/// from the billing addresses when 
		/// both are the same.</remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		/// 
		///		................
		///		//Ship is the object of
		///		//ShipTo populated with 
		///		//the shipping addresses.
		///		................
		///		
		///		
		///		BillTo Bill;
		///		
		///		//Populate billing addresses
		///		//from shipping addresses.
		///		Bill = Ship.Copy();
		///		
		///		................
		/// 
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		/// 
		///		................
		///		'Ship is the object of
		///		'ShipTo populated with 
		///		'the shipping addresses.
		///		................
		///		
		///		
		///		BillTo Bill;
		///		
		///		'Populate billing addresses
		///		'from shipping addresses.
		///		Bill = Ship.Copy()
		///		
		///		................
		/// 
		/// </code>
		/// </example>
		/// <seealso cref="BillTo"/>
		public BillTo Copy()
		{
			BillTo CopyObject = new BillTo();
			CopyObject.AddressCity = AddressCity;
			CopyObject.AddressCountry = AddressCountry;
			CopyObject.AddressEmail = AddressEmail;
			CopyObject.AddressFax = AddressFax;
			CopyObject.AddressFirstName = AddressFirstName;
			CopyObject.AddressLastName = AddressLastName;
			CopyObject.AddressMiddleName = AddressMiddleName;
			CopyObject.AddressPhone2 = AddressPhone2;
			CopyObject.AddressPhone = AddressPhone;
			CopyObject.AddressState = AddressState;
			CopyObject.AddressStreet = AddressStreet;
			CopyObject.AddressStreet2 = AddressStreet2;
			CopyObject.AddressZip = AddressZip;
			return CopyObject;
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal override void GenerateRequest()
		{
			try
			{
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOSTREET, this.ShipToStreet));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOSTREET2, this.ShipToStreet2));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOCITY, this.ShipToCity));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOSTATE, this.ShipToState));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOCOUNTRY, this.ShipToCountry));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOZIP, this.ShipToZip));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOPHONE, this.ShipToPhone));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOPHONE2, this.ShipToPhone2));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOEMAIL, this.ShipToEmail));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOFIRSTNAME, this.ShipToFirstName));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOMIDDLENAME, this.ShipToMiddleName));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPTOLASTNAME, this.ShipToLastName));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPCARRIER, mShipCarrier));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPMETHOD, mShipMethod));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPFROMZIP, mShipFromZip));
				RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_SHIPPEDFROMZIP, mShipFromZip));
			}
			catch (BaseException)
			{
				throw;
			}
			catch (Exception Ex)
			{
				DataObjectException DEx = new DataObjectException(Ex);
				throw DEx;
			}
            //catch
            //{
            //    throw new Exception();				
            //}
		}

		#endregion
	}
}

File: java/src/sdk/dataobjects/paypal/payflow/ShipTo.java
package paypal.payflow;

/**
 * Used for shipping address information
 * <p>Shipping address is destination address information.</p>
 * <p>Following example shows how to use BillTo.</p>
 *
 * .................
 * //inv is the Invoice object.
 * .................
 * <p/>
 * //Set the Shipping Address details.
 * ShipTo ship = new ShipTo();
 * ship.setShipToStreet( "123 Main St.");
 * ship.setShipToZip("12345");
 * inv.setShipTo (ship);
 * .................
 */

public final class ShipTo extends Address {
    private String shipMethod;
    private String shipCarrier;
    private String shipFromZip;

    /**
     * Gets the shipping city
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOCITY</p>
     */
    public String getShipToCity() {
        return super.getAddressCity();
    }

    /**
     * Sets the shipping city
     *
     * @param shipToCity String
     *  <p>Maps to Payflow Parameter: SHIPTOCITY</p>
     */
    public void setShipToCity(String shipToCity) {
        super.setAddressCity(shipToCity);
    }

    /**
     * Gets the Shipping Country
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOCOUNTRY</p>
     */
    public String getShipToCountry() {
        return super.getAddressCountry();
    }

    /**
     * Sets the Shipping Country
     *
     * @param shipToCountry String
     *  <p>Maps to Payflow Parameter: SHIPTOCOUNTRY</p>
     */
    public void setShipToCountry(String shipToCountry) {
        super.setAddressCountry(shipToCountry);
    }

    /**
     * Gets the shipping email.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOEMAIL</p>
     */
    public String getShipToEmail() {
        return super.getAddressEmail();
    }

    /**
     * Sets the shipping email.
     *
     * @param shipToEmail String
     *  <p>Maps to Payflow Parameter: SHIPTOEMAIL</p>
     */
    public void setShipToEmail(String shipToEmail) {
        super.setAddressEmail(shipToEmail);
    }

    /**
     * Gets the shipping first name
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOFIRSTNAME</p>
     */
    public String getShipToFirstName() {
        return super.getAddressFirstName();
    }

    /**
     * Sets the shipping first name
     *
     * @param shipToFirstName String
     *  <p>Maps to Payflow Parameter: SHIPTOFIRSTNAME</p>
     */
    public void setShipToFirstName(String shipToFirstName) {
        super.setAddressFirstName(shipToFirstName);
    }

    /**
     * Gets the last name from the shipping address.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOLASTNAME</p>
     */
    public String getShipToLastName() {
        return super.getAddressLastName();
    }

    /**
     * Sets the last name in the shipping address.
     *
     * @param shipToLastName String
     *  <p>Maps to Payflow Parameter: SHIPTOLASTNAME</p>
     */
    public void setShipToLastName(String shipToLastName) {
        super.setAddressLastName(shipToLastName);
    }

    /**
     * Gets the middlename from the shipping address.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOMIDDLENAME</p>
     */
    public String getShipToMiddleName() {
        return super.getAddressMiddleName();
    }

    /**
     * Sets the middlename in the shipping address.
     *
     * @param shipToMiddleName String
     *  <p>Maps to Payflow Parameter: SHIPTOMIDDLENAME</p>
     */
    public void setShipToMiddleName(String shipToMiddleName) {
        super.setAddressMiddleName(shipToMiddleName);
    }

    /**
     * Gets the Phone2 from the shipping address.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOPHONE2</p>
     */
    public String getShipToPhone2() {
        return super.getAddressPhone2();
    }

    /**
     * Sets the Phone2 in the shipping address.
     *
     * @param shipToPhone2 String
     *  <p>Maps to Payflow Parameter: SHIPTOPHONE2</p>
     */
    public void setShipToPhone2(String shipToPhone2) {
        super.setAddressPhone2(shipToPhone2);
    }

    /**
     * Gets the phone number from the shipping address.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOPHONE</p>
     */
    public String getShipToPhone() {
        return super.getAddressPhone();
    }

    /**
     * Sets the phone number in the shipping address.
     *
     * @param shipToPhone String
     *  <p>Maps to Payflow Parameter: SHIPTOPHONE</p>
     */
    public void setShipToPhone(String shipToPhone) {
        super.setAddressPhone(shipToPhone);
    }

    /**
     * Gets the shipping state.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOSTATE</p>
     */
    public String getShipToState() {
        return super.getAddressState();
    }

    /**
     * Sets the shipping state.
     *
     * @param shipToState String
     *  <p>Maps to Payflow Parameter: SHIPTOSTATE</p>
     */
    public void setShipToState(String shipToState) {
        super.setAddressState(shipToState);
    }

    /**
     * Gets the shipping street.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOSTREET</p>
     */
    public String getShipToStreet() {
        return super.getAddressStreet();
    }

    /**
     * Sets the shipping street.
     *
     * @param shipToStreet String
     *  <p>Maps to Payflow Parameter: SHIPTOSTREET</p>
     */
    public void setShipToStreet(String shipToStreet) {
        super.setAddressStreet(shipToStreet);
    }

    /**
     * Gets the shipping street2.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOSTREET2</p>
     */
    public String getShipToStreet2() {
        return super.getAddressStreet2();
    }

    /**
     * Sets the shipping street2.
     *
     * @param shipToStreet2 String
     *  <p>Maps to Payflow Parameter: SHIPTOSTREET2</p>
     */
    public void setShipToStreet2(String shipToStreet2) {
        super.setAddressStreet2(shipToStreet2);
    }

    /**
     * Gets the shipping zip code.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPTOZIP</p>
     */
    public String getShipToZip() {
        return super.getAddressZip();
    }

    /**
     * Sets the shipping zip code.
     *
     * @param shipToZip String
     *  <p>Maps to Payflow Parameter: SHIPTOZIP.</p>
     */
    public void setShipToZip(String shipToZip) {
        super.setAddressZip(shipToZip);
    }

    /**
     * Gets the shipping method.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPMETHOD</p>
     */
    public String getShipMethod() {
        return shipMethod;
    }

    /**
     * Sets the shipping method.
     *
     * @param shipMethod String
     *  <p>Maps to Payflow Parameter: SHIPMETHOD</p>
     */
    public void setShipMethod(String shipMethod) {
        this.shipMethod = shipMethod;
    }

    /**
     * Gets the shipping carrier.
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPCARRIER</p>
     */
    public String getShipCarrier() {
        return shipCarrier;
    }

    /**
     * Sets the shipping carrier.
     *
     * @param shipCarrier String
     *  <p>Maps to Payflow Parameter: SHIPCARRIER</p>
     */
    public void setShipCarrier(String shipCarrier) {
        this.shipCarrier = shipCarrier;
    }

    /**
     * Gets the ship from zip.
     * <p>Ship from postal code (called ZIP code in the USA).</P>
     *
     * @return String
     *  <p>Maps to Payflow Parameter: SHIPFROMZIP</p>
     */
    public String getShipFromZip() {
        return shipFromZip;
    }

    /**
     * Sets the ship from zip.
     * <p>Ship from postal code (called ZIP code in the USA).</P>
     *
     * @param shipFromZip String
     *  <p>Maps to Payflow Parameter: SHIPFROMZIP</p>
     */
    public void setShipFromZip(String shipFromZip) {
        this.shipFromZip = shipFromZip;
    }

    /**
     * Constructor
     * <p>Shipping address is destination address information.</p>
     * <p>Following example shows how to use ShipTo.</p>
     *
     *  .................
     * //inv is the Invoice object.
     * .................
     * <p/>
     * //Set the Shipping Address details.
     * ShipTo ship = new ShipTo();
     * ship.setShipToStreet ("685A E. Middlefield Rd.");
     * ship.ShipToZip ("94043");
     * inv.setShipTo( ship);
     * .................
     */
    public ShipTo() {
    }

    /**
     * This method copies the common contents
     * from shipping to billing address.
     *
     * @return BillTo
     *         <p>This method can be used to
     *         populate the shipping addresses directly
     *         from the billing addresses when
     *         both are the same.</p>
     *  ................
     * //ship is the object of
     * //shipTo populated with
     * //the shipping addresses.
     * ................
     * <p/>
     * <p/>
     * BillTo bill;
     * <p/>
     * //Populate billing addresses
     * //from shipping addresses.
     * bill = ship.copy();
     * <p/>
     * ................
     */
    public BillTo copy() {
        BillTo copyObject = new BillTo();
        copyObject.setAddressCity(this.getAddressCity());
        copyObject.setAddressCountry(this.getAddressCountry());
        copyObject.setAddressEmail(this.getAddressEmail());
        copyObject.setAddressFax(this.getAddressFax());
        copyObject.setAddressFirstName(this.getAddressFirstName());
        copyObject.setAddressLastName(this.getAddressLastName());
        copyObject.setAddressMiddleName(this.getAddressMiddleName());
        copyObject.setAddressPhone2(this.getAddressPhone2());
        copyObject.setAddressPhone(this.getAddressPhone());
        copyObject.setAddressState(this.getAddressState());
        copyObject.setAddressStreet(this.getAddressStreet());
        copyObject.setAddressStreet2(this.getAddressStreet2());
        copyObject.setAddressZip(this.getAddressZip());
        return copyObject;
    }

    protected void generateRequest() {
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOSTREET, this.getShipToStreet()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOSTREET2, this.getShipToStreet2()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOCITY, this.getShipToCity()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOSTATE, this.getShipToState()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOCOUNTRY, this.getShipToCountry()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOZIP, this.getShipToZip()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOPHONE, this.getShipToPhone()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOPHONE2, this.getShipToPhone2()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOEMAIL, this.getShipToEmail()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOFIRSTNAME, this.getShipToFirstName()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOMIDDLENAME, this.getShipToMiddleName()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPTOLASTNAME, this.getShipToLastName()));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPCARRIER, shipCarrier));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPMETHOD, shipMethod));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPFROMZIP, shipFromZip));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_SHIPPEDFROMZIP, shipFromZip));
    }

}
